ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -315,7 +375,7 @@ private boolean isExecutableDdlStatement(final Statement statement) {
         && !((statement instanceof SetProperty) || (statement instanceof UnsetProperty));
   }
 
-  private CommandStatusEntity distributeStatement(
+  private synchronized CommandStatusEntity distributeStatement(
```

Comment:
Why are we synchronizing this call?

Doesn't this call block waiting for a response?  That would mean each KSQL instance could only handle one pending request at a time...

Reply:
It was to prevent adding statements and terminating cluster concurrently write into the command topic. With adding of `acceptingStatements` to the engine no need for this. Removed it.